src/InputText/internal/InputText.tsx

```diff
@@ -224,25 +226,38 @@ export const InputText = forwardRef<HTMLInputElement | HTMLTextAreaElement, Inpu
       <Align horizontal center hug="width">
         <Spacer xsmall={p.size === "small"} small={p.size === "large"} />
 
-        <Icon
-          name={p.iconNameLeft}
-          medium={p.size === "small"}
-          large={p.size === "large"}
-          fill={[p.error ? Color.Error : p.color, 700]}
-        />
+        {p.onIconClick ? (
+          <InputButtonIconTertiary hug iconName={p.iconNameLeft} size={p.size} onClick={() => p.onIconClick!("left")} />
```

Comment:
The `!` operator should not be necessary here.

```suggestion
          <InputButtonIconTertiary hug iconName={p.iconNameLeft} size={p.size} onClick={() => p.onIconClick("left")} />
```